src/Service/Terminal/Service_Terminal.re

```diff
@@ -63,14 +96,25 @@ module Sub = {
             Internal.onExtensionMessage, (msg: ExtHostClient.Terminal.msg) => {
             switch (msg) {
             | SendProcessTitle({terminalId, title}) =>
-              dispatch(ProcessTitleChanged({id: terminalId, title}))
+              dispatchIfMatches(
+                terminalId,
+                ProcessTitleChanged({id: terminalId, title}),
+              )
             | SendProcessPid({terminalId, pid}) =>
-              dispatch(ProcessStarted({id: terminalId, pid}))
+              dispatchIfMatches(
+                terminalId,
+                ProcessStarted({id: terminalId, pid}),
+              )
+            | SendProcessData({terminalId, data}) =>
+              if (terminalId == params.id) {
+                let _: int = ReveryTerminal.write(~input=data, terminal);
+                ();
```

Comment:
You can also do `ReveryTerminal.write(~input=data, terminal) |> (ignore: int => unit);`. Not saying that's better, but might be, considering `refmt`s behaviour.

What is the ìnt` anyway?  Chars written? I find the API is usually better if it has a separate function returning some status value if that's needed (which it almost never is), while the "plain" one just returns `unit`.

Reply:
Indeed, it's returning the chars actually written. I agree, we should be able to tuck this completely under the `ReveryTerminal` abstraction - https://github.com/revery-ui/revery-terminal/pull/5

Reply:
Fixed this now 👍